### PR TITLE
update hunter formular and JS controller

### PR DESCRIPTION
**Pull request by erikmacej**

### CODE REVIEW

File: web/src/main/webapp/admin_hunter_controllers.js [hunk 1]
```diff
@@ -35,4 +35,46 @@ portalControllers.controller('AdminHuntersCtrl',
             );
         };
 
+    });
+
+function findHunterById($hunterId, $scope, $http) {
+    $http.get('/pa165/rest/hunter/' + $hunterId).then(function (response) {
+        $scope.hunter = response.data;
+        console.log('deleted hunter ' + hunter.id + ' (' + hunter.email + ')');
```
Comment (atiratree) [inline web/src/main/webapp/admin_hunter_controllers.js]: please remove or change this log,
it is throwing  angular.js:12477 ReferenceError: hunter is not defined
and it is about find, not delete

File: web/src/main/webapp/resources/partials/forms/update_hunter.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            <p class="help-block" ng-show="form.firstName.$error.minlength">First name must be at least 3 characters</p>
+            <p class="help-block" ng-show="form.firstName.$error.maxlength">First name must be at shorter than 50 characters</p>
+        </div>
+    </div>
+
+    <div class="form-group" ng-class="{'has-error': !(form.surname.$valid)}">
+        <label for="surname" class="col-sm-2 control-label">Surname</label>
+        <div class="col-sm-10">
+            <input placeholder="surname" id="surname" type="text" ng-minlength="3" ng-maxlength="50" name="surname" class="form-control" ng-model="hunter.surname" />
+            <p class="help-block" ng-show="form.surname.$error.minlength">Surname must be at least 3 characters</p>
```
Comment (atiratree) [inline web/src/main/webapp/resources/partials/forms/update_hunter.html]: I guess all of these constraints should be also defined on server. It is not really safe when you can call rest api to circumvent these checks

File: web/src/main/webapp/resources/partials/forms/update_hunter.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <label for="rank" class="col-sm-2 control-label">Rank</label>
+        <div class="col-sm-10">
+            <select id="rank" name="rank" class="form-control" ng-model="hunter.rank">
+                <option ng-repeat="rank in ranks" value="{{rank}}">{{rank}}</option>
+            </select>
+        </div>
+    </div>
+
+    <button ng-disabled="!form.$valid" class="btn btn-success" type="submit" ng-click="updateHunter(hunter)">Update Hunter</button>
+    <a class="btn btn-danger" href="#/admin/hunters" role="button">Back</a>
```
Comment (atiratree) [inline web/src/main/webapp/resources/partials/forms/update_hunter.html:54]: Back button is usually on the left :) . Maybe it disturbs just me.

**Comment by atiratree** [comment]
maybe even the order of Update | Delete buttons could be reversed

File: web/src/main/webapp/resources/partials/forms/update_hunter.html (hunk 2 above)
Comment (erikmacej) [reply web/src/main/webapp/resources/partials/forms/update_hunter.html]: Problem is with @Size annotation. Surname and firstname is not required, When you submit form with no name or surname you don't send null strings but empty strings ("" => size=0). And validation throw constraint violation.

**Pull request merged**